services/st_util/StBootloaderCommunicatorUart.cpp

```diff
@@ -314,9 +318,9 @@ namespace services
             onError(reason);
     }
 
-    void StBootloaderCommunicatorUart::SendData(infra::ConstByteRange data, uint8_t checksum)
+    void StBootloaderCommunicatorUart::SendData(infra::ConstByteRange data, const uint8_t& checksum)
```

Comment:
Why is this now a reference? This implies that the sending side now has to keep the checksum alive a lot longer

Reply:
Probably because you're using SendData(infra::MakeConstByteRange(checksum)) below. I would instead copy the checksum to a data member and avoid burdening the calling site